### PR TITLE
Fire onMouseEnter/Leave events on pointer concluded

### LINKED ISSUE #8525: onMouseEnter / onMouseLeave does not fire on pointer concluded
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Open RNTester > Mouse Events example
2. Click from inside a box, while holding, move to outside a box, release click without moving mouse
3. onMouseLeave is not called on the box you left

Similar behavior occurs if you click from outside a box and conclude the gesture inside the box.

## Expected Results
The onMouseEnter / onMouseLeave state should update

## Snack, code example, screenshot, or link to a repository:

https://user-images.githubusercontent.com/1106239/131731777-2ed69744-78d8-43c5-9776-e3c3c05f5058.mp4

### PR DESCRIPTION
**Pull request by rozele**
See details of bug in #8525. Calling UpdatePointersInViews in the OnPointerConcluded callback fixes the issue.

Fixes #8525

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8526)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp [hunk 1]
```diff
@@ -194,6 +194,8 @@ void TouchEventHandler::OnPointerConcluded(TouchEventType eventType, const winrt
     m_touchId = 0;
 
   m_xamlView.as<xaml::FrameworkElement>().ReleasePointerCapture(args.Pointer());
+
+  UpdatePointersInViews(args, tag, sourceElement);
```
Comment (rozele) [inline vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp:198]: There are probably some edge cases we want to check for here. For example, we do fire UpdatePointersInViews after capturing the pointer in OnPointerPressed. Perhaps we should check that we had previously captured the pointer before calling UpdatePointersInViews?

Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp:198]: For example, this seems to work:
```
  const auto frameworkElement = m_xamlView.as<xaml::FrameworkElement>();
  const auto wasCaptured =
      frameworkElement.PointerCaptures() != nullptr && frameworkElement.PointerCaptures().Size() > 0;

  m_xamlView.as<xaml::FrameworkElement>().ReleasePointerCapture(args.Pointer());

  // Updates the enter/leave state when pointer was previously captured
  if (wasCaptured) {
    UpdatePointersInViews(args, tag, sourceElement);
  }
```

**Review by NickGerleman — APPROVED** [review]

**Comment by NickGerleman** [comment]
@msftbot merge in 12 hours

**Comment by rozele** [comment]
Thanks @NickGerleman - can you also restart CI (I don't have permissions). I believe the current failure is a flaky build.

**Pull request merged**